File: Nível 1/1515 - Hello Galaxy/main.lua
repeat
    quantia = tonumber(io.read())
    if(quantia == 0) then break end 
    
    menor = 99999999999999
    civilizacao = "nenhum"

    for i = 1, quantia do
        infos = io.read()
        nome, ano, tempo = infos:match("(%S+)%s+(%S+)%s+(%S+)")
        ano = tonumber(ano)
        tempo = tonumber(tempo)

        calc = ano - tempo 

        if(calc < menor) then 
            menor = calc 
            civilizacao = nome
        end
    end

    print(civilizacao)
until false

File: Nível 1/1515 - Hello Galaxy/main.py
while(True):
    quantia = int(input("A: "))
    if(quantia == 0): break 

    menor = 999999999999999
    civilizacao = "nenhum"

    for i in range(1, quantia + 1):
        nome, ano, tempo = map(str, input("B: ").split(" "))
        ano = int(ano)
        tempo = int(tempo)

        calc = ano - tempo 

        if(calc < menor):
            menor = calc
            civilizacao = nome
    
    print(civilizacao)

